Source/WebKit/Platform/IPC/ArgumentCoders.h

```diff
@@ -782,4 +782,32 @@ template<> struct ArgumentCoder<std::nullptr_t> {
     static std::optional<std::nullptr_t> decode(Decoder&) { return nullptr; }
 };
 
+template<typename T, typename Traits> struct ArgumentCoder<WTF::Markable<T, Traits>> {
+    template<typename Encoder, typename U>
+    static void encode(Encoder& encoder, U&& markable)
+    {
+        bool isEmpty = !markable;
+        encoder << isEmpty;
+        if (!isEmpty)
+            encoder << *markable;
+    }
+
+    template<typename Decoder>
+    static std::optional<WTF::Markable<T>> decode(Decoder& decoder)
+    {
+        auto isEmpty = decoder.template decode<bool>();
+        if (!isEmpty)
+            return std::nullopt;
+
+        if (*isEmpty)
+            return WTF::Markable<T, Traits> { };
+
+        auto value = decoder.template decode<T>();
+        if (!value)
+            return std::nullopt;
+
+        return WTF::Markable<T, Traits>(WTFMove(*value));
```

Comment:
I think this should strictly speaking be:
`*WTFMove(value)`.
This would call `T&& std::optional<T>::operator*() &&;`  and the result is rvalue reference of `T`, which you want.

Consider you have `T == int`. You cannot `std::move()` an int.
Consider you have `T == MyClass`. You must `std::move()` the MyClass to prevent copy.

You cannot `std::forward()` as `value` is not a forwarding reference.

Reply:
Right, I missed that. Thank you!

Reply:
Hm, checking the other implementations `WTFMove(*value)` is used.

Reply:
I'm gonna stick to what was in the original patch, as I see that's what's done for other serializers and the original implementation as well.